### DISCUSSION TITLE
Locked / unpaired the right hand side of a split keyboard

### DISCUSSION THREAD

**Discussion by lukewilde** [manna-harbour/miryoku#211] [Q&A] [answered]
Last night I was fudging some media shortcuts and after mashing a few incorrect key combos the right side (slave) of my keyboard has totally stopped responding to key presses. I was being clumbsy, so it's entirely possible I was mashing "Additional Features" keys or another layer.

My Setup:
- I use ZMK with a nice!nano v2
- Only connected to host over USB and each half is connected via the TRRS cable, no bluetooth
- Corne shield
- [Here's the customisations I've made to my firmware](https://github.com/manna-harbour/miryoku_zmk/compare/master...lukewilde:miryoku_zmk:master)

I've tried:
- Re-flashing the firmware
- Flashed the [Reset settings firmware](https://zmk.dev/docs/troubleshooting#split-keyboard-halves-unable-to-pair) and completed the steps to re-pair
- I flashed the left.uf2 onto the malfunctioning right side and after connecting it via USB to my host, it functions correctly (but with mirrored keys)
  - I then flashed the right.uf2 onto the other side and in this configuration both sides actually paired and output key presses, but are mirrored as expected

I've pulled the board apart looking for shorts and bad solder joints and don't think it's a hardware issue, something which I think is supported by both sides working after I switched the left / right firmware.

Any ideas?

**Comment by manna-harbour** [comment, marked as answer]
There are no Miryoku-specific persistent settings, so it won't be a Miryoku-specific issue, and the best place for help is the help channel on the ZMK discord.

You can lock layers, but it requires a quick double tap so is unlikely to be done by accident, and the keyboard will always start from Base after reset.

Usually a settings reset and reflash will fix any issues with split comms. Make sure the firmware for both sides are built from the same source. It's unlikely to be the issue, but you could also try building from GitHub actions to eliminate any issues with the local build environment. There is also a full erase firmware available on the discord that might be worth trying.

If the peripheral is not paired with the central you won't get any keystrokes from the peripheral, so try settings reset and re-pairing of sides again, taking care to flash the correct firmware to each side and carefully following the split pairing procedure.

Successfully swapping firmware between sides suggests a hardware issue. If your controllers are socketed you can also try swapping controllers between sides, and also running the bare MCUs removed from the PCBs and testing by shorting pins. 

Nice!Nano is susceptible to internal shorts, possibly caused by static discharge. Also, split comms is always via Bluetooth and use of TRRS cables can cause damage.

You can build with modified Kconfig to set right central, so there won't be keymap mirroring. (Another way is to unmirror with a custom mapping, but the Kconfig method would be better.)

Also, you could try USB logging.

**Reply by manna-harbour** [reply]
This is the full flash erase firmware: https://github.com/joric/zmk/releases/download/jorne-0.02/erase_flash.ino.uf2.

**Reply by manna-harbour** [reply]
In general, also try the default keymap, but I don't think it will help in this case.

**Reply by lukewilde** [reply]
Amazing, thank you so much for the detailed response! I'll let you know how I get on 😁
